package.json

```
@@ -21,18 +22,29 @@
     "bignumber.js": "^9.0.0"
   },
   "devDependencies": {
+    "@babel/cli": "^7.19.3",
+    "@babel/core": "^7.19.6",
+    "@babel/polyfill": "^7.12.1",
+    "@babel/preset-env": "^7.19.4",
+    "@babel/preset-typescript": "^7.18.6",
+    "@goodeggs/toolkit": "^7.0.1",
+    "@goodeggs/tsconfig": "^2.0.2",
     "chai": "^4.2.0",
-    "coffee-script": ">=1.8.x",
     "mocha": "^7.1.0"
   },
   "publishConfig": {
     "registry": "https://registry.npmjs.org/",
     "always-auth": true
   },
   "scripts": {
-    "compile": "coffee --bare --compile --output lib/ src/",
-    "prepublish": "npm run compile",
-    "pretest": "npm run compile",
-    "test": "mocha"
+    "prepublishOnly": "yarn run clean && yarn run build",
+    "build": "babel --extensions .ts,.js --source-maps --ignore 'src/*.test.ts' --out-dir lib/ src/",
```

nitpick, clean is a part of build
```suggestion
    "prepublishOnly": "yarn run build",
    "build": "yarn run clean && yarn run build:es",
    "build:es": "babel --extensions .ts,.js --source-maps --ignore 'src/*.test.ts' --out-dir lib/ src/",
```